### ISSUE TITLE
Report coverage outside of buildbot even on errors

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8598]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8598|
|Type|enhancement|
|Created|2016-07-13 18:42:02Z|
|Branch|https://github.com/twisted/twisted/tree/8598-coverage-report-on-failure|

Right now for travis (and soon Appveyor) the coverate is reported from tox.

... but if a test fails the coverage is no longer reported...

This should be fine, but since a coverage report is missing, codecov.io  will not report the final coverage status.

We need to always send a coverage report from a builder to let codecov.io know that we are done.

<details><summary>Searchable metadata</summary>

```
trac-id__8598 8598
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8598_coverage_report_on_failure 8598-coverage-report-on-failure
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1468435322729394 1468435322729394
changetime__1468718576287296 1468718576287296
version__None None
owner__adiroiban adiroiban

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban removed owner|
|-|-|

Here is the PR

https://github.com/twisted/twisted/pull/328

please check if it make sense.

it uses 2 tox environments... one to run the tests and another one to publish them to codecov... in this way publishing is always called, even if tests are failing.

The coverage raw reports  are not stored in the tox workdir.

**Comment by twisted-trac** [comment]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)<a name="note_2"></a>|@markrwilliams commented|
|-|-|

The configuration changes are reasonable and now the coverage reports are coming through.

The buildbot failures are apparently unrelated to this PR (e.g. twistedchecker failures in unrelated files and various build bots that fail here and in other PRs as well.)

This looks ready for merging to me.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @adiroiban<br>@adiroiban set status to `assigned`|
|-|-|

Will merge. Thanks!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset 7ed6240b79aa3586e0e1d620109297d5c5330d79
```
#!CommitTicketReference repository="" revision="7ed6240b79aa3586e0e1d620109297d5c5330d79"
Merge 8598-coverage-report-on-failure: Publish coverage even on test failure.


Author: adiroiban
Reviewer: markrwilliams
Fixes: #8598
```
